File: Server/ChatApp/ChatApp.Backend/Core/Conversations/DTOs/ConversationListDTO.cs
using ChatApp.Backend.Core.Enums;
using ChatApp.Backend.Core.Messages.DTOs;
using ChatApp.Backend.Core.Users.DTOs;
using ChatApp.Backend.Domain;

namespace ChatApp.Backend.Core.Conversations.DTOs;

public record ConversationListDto(
    int Id,
    ConversationType Type,
    List<UserDto> Members,
    LastMessageDto? LastMessage
);

File: Front/lib/data/dto/conversations/conversation_list_dto.dart
import 'package:chatapp_frontend/core/enums/conversation_type.dart';
import 'package:chatapp_frontend/data/dto/messages/last_message_dto.dart';
import 'package:chatapp_frontend/data/dto/users/user_dto.dart';
import 'package:json_annotation/json_annotation.dart';

part 'conversation_list_dto.g.dart';

@JsonSerializable()
class ConversationListDto {
  ConversationListDto({
    required this.id,
    required this.type,
    required this.members,
    required this.currentUser,
    required this.lastMessage,
  });

  factory ConversationListDto.fromJson(Map<String, dynamic> json) =>
      _$ConversationListDtoFromJson(json);

  Map<String, dynamic> toJson() => _$ConversationListDtoToJson(this);

  final int id;
  final ConversationType type;
  final List<UserDto> members;
  final UserDto currentUser;
  final LastMessageDto? lastMessage;
}
